src/Mod/AddonManager/addonmanager_utilities.py

```diff
@@ -417,14 +418,25 @@ def run_interruptable_subprocess(args) -> subprocess.CompletedProcess:
     stdout = ""
     stderr = ""
     return_code = None
+    counter = 0
     while return_code is None:
+        counter += 1
         try:
-            stdout, stderr = p.communicate(timeout=10)
+            stdout, stderr = p.communicate(
+                timeout=1
+            )  # one second timeout allows interrupting the run once per second
             return_code = p.returncode
         except subprocess.TimeoutExpired:
-            if QtCore.QThread.currentThread().isInterruptionRequested():
+            if (
+                hasattr(QtCore, "QThread")
+                and QtCore.QThread.currentThread().isInterruptionRequested()
+            ):
                 p.kill()
                 raise ProcessInterrupted()
+            if counter >= timeout_secs:  # The real timeout
+                p.kill()
+                stdout, stderr = p.communicate()
+                return_code = -1
```

Comment:
The code could be simplified by removing the counter and just look at the elapsed time instead.
```python
if time.time() - start_time >= timeout_secs
```